### PR TITLE
Time as an input with tests

### LINKED ISSUE #248: Unable to use scalar Time for Inputs
It looks like when you try to use scalar time as an input, you get an error message "cant unmarshal <string> (string) as time.Time 

screen shot 2018-08-16 at 9 54 03 pm

### PR DESCRIPTION
**Pull request by Melaninneal**
Please consider adding the ability to have type "time" for inputs fixes #248

### CODE REVIEW

File: internal/exec/packer/packer.go [hunk 1]
```diff
@@ -325,7 +325,10 @@ func unmarshalInput(typ reflect.Type, input interface{}) (interface{}, error) {
 	if reflect.TypeOf(input) == typ {
 		return input, nil
 	}
-
+	if typ.String() == "time.Time" {
```
Comment (vetcher) [inline internal/exec/packer/packer.go]: may be we can make it in more flexible way?
what if I want to import package `time` like? 
```go
import . "time"
```
or 
```go
import stdtime "time"
```

Comment (pavelnikolov) [reply internal/exec/packer/packer.go]: No, I wouldn't add alias to standard library packages and I would definitely avoid `.` imports. https://github.com/golang/go/wiki/CodeReviewComments#import-dot

File: internal/exec/packer/packer.go [hunk 2]
```diff
@@ -359,6 +362,10 @@ func unmarshalInput(typ reflect.Type, input interface{}) (interface{}, error) {
 	return nil, fmt.Errorf("incompatible type")
 }
 
+func convertStringToTime(timeString string) (interface{}, error){
```
Comment (pavelnikolov) [inline internal/exec/packer/packer.go]: Do you need to return `interface{}` here? Can't your return time?

File: internal/exec/packer/packer.go (hunk 1 above)
Comment (pavelnikolov) [reply internal/exec/packer/packer.go]: Also:
> Avoid renaming imports except to avoid a name collision;...

https://github.com/golang/go/wiki/CodeReviewComments#imports

File: internal/exec/packer/packer.go [hunk 3]
```diff
@@ -325,7 +325,10 @@ func unmarshalInput(typ reflect.Type, input interface{}) (interface{}, error) {
 	if reflect.TypeOf(input) == typ {
 		return input, nil
 	}
-
+	if typ.String() == "time.Time" {
+		return convertStringToTime(input.(string))
```
[the remark this answers is not in the document]
Comment (Melaninneal) [reply internal/exec/packer/packer.go]: i realized this doesn't quite work with Nullable time objects will add second condition and remove convertStringToTime func

**Comment by BarthesSimpson** [comment]
@pavelnikolov - any chance this can be merged now that the convertStringToTime function has been removed?

File: internal/exec/packer/packer.go [hunk 4]
```diff
@@ -5,10 +5,10 @@ import (
 	"math"
 	"reflect"
 	"strings"
-
-	"github.com/graph-gophers/graphql-go/errors"
+		"github.com/graph-gophers/graphql-go/errors"
 	"github.com/graph-gophers/graphql-go/internal/common"
 	"github.com/graph-gophers/graphql-go/internal/schema"
+	"time"
```
Comment (pavelnikolov) [inline internal/exec/packer/packer.go]: Please, fix the imports order

File: internal/exec/packer/packer.go [hunk 5]
```diff
@@ -359,6 +368,8 @@ func unmarshalInput(typ reflect.Type, input interface{}) (interface{}, error) {
 	return nil, fmt.Errorf("incompatible type")
 }
 
+
+
```
Comment (pavelnikolov) [inline internal/exec/packer/packer.go]: Unrelated change. Please, remove these empty lines.

File: graphql_test.go [hunk 6]
```diff
@@ -1691,6 +1706,8 @@ func TestInput(t *testing.T) {
 					recursive(value: {next: {next: {}}})
 					intID: id(value: 1234)
 					strID: id(value: "1234")
+					timeInput: inputTime(value: {t : "2018-08-22T11:34:55Z"})
+					nullableTimeInput: nullableInputTime(value: {t : "2018-08-22T11:34:55Z"})
```
Comment (pavelnikolov) [inline graphql_test.go]: Please, add unit tests with a `null` value.

File: internal/exec/packer/packer.go [hunk 7]
```diff
@@ -325,7 +325,16 @@ func unmarshalInput(typ reflect.Type, input interface{}) (interface{}, error) {
...
+	switch typ.String(){
+	case  "time.Time": {
+		return time.Parse(time.RFC3339, input.(string))
+	}
+	case "*time.Time":
+		t, err := time.Parse(time.RFC3339, input.(string))
+		return &t, err
+	}
+	
+
```
Comment (pavelnikolov) [inline internal/exec/packer/packer.go]: you don't really need double empty lines, no?

**Comment by pavelnikolov** [comment]
Please, add missing unit tests unit tests.

**Comment by pavelnikolov** [comment]
Can you just use `graphql.Time` as the type of your argument? You'll need to define `scalar Time` in your schema in order to be able to use it though.

**Comment by pavelnikolov** [comment]
The underlying issue have been fixed already with a different PR.
